py/picca/cf.py

```diff
@@ -112,7 +112,8 @@ def fast_cf(z1,r1,w1,d1,z2,r2,w2,d2,ang,same_half_plate):
     bt = (rt/rt_max*nt).astype(int)
     bins = bt + nt*bp
     if same_half_plate:
-        w = bp == 0
+        if x_correlation : w = bp == np/2
+        else : w = bp == 0
```

Comment:
this assumes that rp_min = 0 (there's no guarantee that a user won't set rp_min for an auto-correlation)